### PR TITLE
Update while_v2 design doc

### PR DESCRIPTION
**Pull request by skye**

### CODE REVIEW

**Comment by skye** [comment]
@srbs please review

**Comment by skye** [comment]
Note that I rearranged two of the sections so that the overall gradients algorithm comes before the accumulating intermediates section. Sorry to screw up the diff... it's a good chance to revisit the entire gradients section ;)

**Comment by srbs** [comment]
@skye I don't think you meant me

**Comment by skye** [comment]
Ah thanks for letting me know! @saxenasaurabh please review :)

**Comment by ewilderj** [comment]
Pinging @saxenasaurabh

File: rfcs/20180821-differentiable-functional-while.md [hunk 1]
```diff
@@ -65,22 +43,37 @@ Excerpt from white paper on [Control Flow in TensorFlow](http://download.tensorf
...
+
+To get a count of the number of forward iterations we add an integer counter every while_loop which is initialized to 0 and is incremented in the loop body. Note that we just need the total number of iterations for the gradient pass so we do not need to accumulate the intermediate values of the counter. This counter is always the first output of the While op.
```
Comment (saxenasaurabh) [inline rfcs/20180821-differentiable-functional-while.md]: nit: counter "to" every while_loop

File: rfcs/20180821-differentiable-functional-while.md [hunk 2]
```diff
@@ -65,22 +43,37 @@ Excerpt from white paper on [Control Flow in TensorFlow](http://download.tensorf
...
+
+Note that the While [gradient function](https://github.com/tensorflow/tensorflow/blob/0440ccfc199cbffc10aae19fde07f0100c823ed9/tensorflow/python/framework/ops.py#L2352) assumes that the first loop output is the
```
Comment (saxenasaurabh) [inline rfcs/20180821-differentiable-functional-while.md]: The link here points to `def RegisterGradient`. Is that intentional? Maybe point to [_WhileGrad](https://github.com/tensorflow/tensorflow/blob/ec1effdb69d33c947f30a5155c5cc4104c07a87e/tensorflow/python/ops/while_v2.py#L248)

**Comment by ewilderj** [comment]
Thanks! @skye if you push a fix to the nits I'll go ahead and get this merged. Thank you!

File: rfcs/20180821-differentiable-functional-while.md (hunk 1 above)
Comment (skye) [reply rfcs/20180821-differentiable-functional-while.md]: Done.

File: rfcs/20180821-differentiable-functional-while.md (hunk 2 above)
Comment (skye) [reply rfcs/20180821-differentiable-functional-while.md]: I must have thought that made sense but I can't understand why now :) Changed to the _WhileGrad link.

**Comment by skye** [comment]
Just pushed a fix, thanks @saxenasaurabh and @ewilderj !

**Pull request merged**